From a6d60ff3dd540cc5a7e26dcbdd267fb2c17681bf Mon Sep 17 00:00:00 2001
From: chdugw_auto <jkulkarni@maxlinear.com>
Date: Tue, 24 Aug 2021 11:52:51 +0300
Subject: [PATCH] Merge pull request #3492 in SW_UGW/linux from
 feature/DRV_DPM_SW-528-support-codel-parameter-for-qos-queue-apis-flm to
 master

* commit '1426c4993d6e921765b4ee2d3019b8c59433a414':
  DRV_DPM_SW-528 - support codel parameter for QoS Queue
---
 include/net/datapath_api_qos.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/net/datapath_api_qos.h b/include/net/datapath_api_qos.h
index 41abb953f35a..655818660d63 100644
--- a/include/net/datapath_api_qos.h
+++ b/include/net/datapath_api_qos.h
@@ -742,6 +742,12 @@ enum dp_q_size_unit {
 	DP_COLOR_BYTE,   /*!< in bytes*/
 };
 
+/*! @brief dp_q_codel */
+enum dp_q_codel {
+	DP_CODEL_DIS = 0, /*!< codel disable */
+	DP_CODEL_EN,   /*!< codel enable */
+};
+
 /*! @brief dp_queue_conf*/
 struct dp_queue_conf {
 	int inst; /*!< input: dp instance. For SOC side, it is always zero */
@@ -759,6 +765,7 @@ struct dp_queue_conf {
 	u32 wred_slope[DP_MAX_COLORS];/*!< in percent, for example, 1 means 1%*/
 	u32 wred_min_guaranteed; /*!< ??? from ppv4 */
 	u32 wred_max_allowed; /*!< ??? from ppv4 */
+	enum dp_q_codel codel;  /*!< [in] enable/disable/ queue codel param */
 };
 
 /*! \addtogroup APIs_dp_queue_conf
